Packs/TOPdesk/Integrations/TOPdesk/TOPdesk.py

```diff
@@ -0,0 +1,862 @@
+        readable_attachment[0][header] = response.get(half_camelized_header, None)
+
+    readable_output = tableToMarkdown(f"{INTEGRATION_NAME} attachment of incident {args.get('number', None)}",
+                                      readable_attachment,
+                                      headers=headers,
+                                      removeNull=True)
+
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix=f'{INTEGRATION_NAME}.attachment',
+        outputs_key_field='id',
+        outputs=response
+    )
+
+
+''' FETCH & MIRRORING COMMANDS'''
+
+
+def fetch_incidents(client: Client,
+                    last_run: Dict[str, Any],
+                    demisto_params: Dict[str, Any]) -> Tuple[Dict[str, Any], List[Dict[str, Any]]]:
+    """Fetches incidents from TOPdesk."""
+
+    first_fetch_datetime = dateparser.parse(demisto_params.get('first_fetch', '3 days'))
+    last_fetch = last_run.get('last_fetch', None)
+
+    if not last_fetch:
+        if first_fetch_datetime:
+            last_fetch_datetime = first_fetch_datetime
+        else:
+            raise Exception("Could not find last fetch time.")
+    else:
+        last_fetch_datetime = dateparser.parse(last_fetch)
+
+    latest_created_time = last_fetch_datetime
+    incidents: List[Dict[str, Any]] = []
+
+    topdesk_incidents = get_incidents_with_pagination(client=client,
+                                                      max_fetch=int(demisto_params.get('max_fetch', 10)),
+                                                      query=demisto_params.get('query', None),
+                                                      modification_date_start=demisto_params.get(
+                                                          'modification_date_start', None),
+                                                      modification_date_end=demisto_params.get(
+                                                          'modification_date_end', None))
+
+    for topdesk_incident in topdesk_incidents:
+        if topdesk_incident.get('creationDate', None):
+            creation_datetime = dateparser.parse(topdesk_incident['creationDate'])
+            incident_created_time = creation_datetime
+        else:
+            incident_created_time = last_fetch_datetime
+        if int(last_fetch_datetime.timestamp()) < int(incident_created_time.timestamp()):
+            labels: List[Dict[str, Any]] = []
+            for topdesk_incident_field, topdesk_incident_value in topdesk_incident.items():
+                if isinstance(topdesk_incident_value, str):
+                    labels.append({
+                        'type': topdesk_incident_field,
+                        'value': topdesk_incident_value
+                    })
+                else:
+                    labels.append({
+                        'type': topdesk_incident_field,
+                        'value': json.dumps(topdesk_incident_value)
+                    })
```

Comment:
If I get where you are aiming, this is used to populate the incident fields right?
if so then there is really no need for this, the incoming mapper should do the trick by taking the rawJSON and using it to map to the fields.

Reply:
This was actually copied from ServiceNow, I thought that the mapper uses these labels in the first place. I'll try it your way, see what happens.